internal/pkg/accrual/client.go

```diff
@@ -59,28 +62,44 @@ func (acc *AccrualClient) Calc(ctx context.Context, number string) (*ResponseOrd
 	client := http.Client{}
 	w, err := client.Do(r)
 	if err != nil {
-		acc.logger.Error(err.Error())
 		return nil, err
 	}
-	if err = checkCalcResponse(w); err != nil {
-		acc.logger.Warn("accrual client has error: " + err.Error())
+	err = checkCalcResponse(w)
+	if errors.Is(err, ErrTooManyRequests) {
+		if tm, errTm := getTimeout(w); err != nil {
+			return nil, errTm
+		} else {
+			return &ResponseOrders{Timeout: tm}, err
+		}
+	}
+	if err != nil {
 		return nil, err
 	}
-	defer w.Body.Close()
 	data := ResponseOrders{}
 	if err := json.NewDecoder(w.Body).Decode(&data); err != nil {
-		acc.logger.Error(err.Error())
 		return nil, err
 	}
+	defer w.Body.Close()
```

Comment:
```suggestion
	w.Body.Close()
```
defer имеет смысл только при наличии еще каких-то вычислений (с возможным экстренным выходом по ошибке) между его определением и концом функции. Поставленный перед концом функции он лишь добавляет временное пенальти для создания отложенной функции.